release/build/main.go

```diff
@@ -206,13 +270,32 @@ func hashreleaseSubCommands(cfg *config.Config, runner *registry.DockerRunner) [
 					return fmt.Errorf("%s must be set if %s is set", skipImageScanFlag, skipValidationFlag)
 				}
 
+				// Extract the version from pinned-version.yaml.
+				hash, err := hashrelease.RetrievePinnedVersionHash(cfg.TmpFolderPath())
+				if err != nil {
+					return err
+				}
+
+				// Check if the hashrelease has already been published.
+				if published := tasks.HashreleasePublished(cfg, hash); published {
+					return fmt.Errorf("hashrelease %s has already been published", hash)
+				}
+
 				// Push the operator hashrelease first before validaion
 				// This is because validation checks all images exists and sends to Image Scan Service
-				tasks.OperatorHashreleasePush(runner, cfg)
+				o := operator.NewManager(
+					operator.WithRepoRoot(cfg.Operator.Dir),
```

Comment:
WithRepoRoot is still the wrong name for this - I think it should be `WithOperatorDirectory`.

"RepoRoot" is referring to the root of the calico repository.